migrate SimpleDialog.input to material TextInputLayout

moving-bits opened the pull request:
## Description
migrate SimpleDialog.input to material TextInputLayout

![image](https://user-images.githubusercontent.com/3754370/122285704-585e1b00-ceef-11eb-96aa-73caad29d0f8.png).![image](https://user-images.githubusercontent.com/3754370/122285725-5d22cf00-ceef-11eb-8ccc-8102ef269568.png).![image](https://user-images.githubusercontent.com/3754370/122285753-62801980-ceef-11eb-964d-61908663691e.png)

The last one is not yet perfect, although it's using identical source code, but as settings are not based on `AppCompatActivity` I guess there's not much we can do about it.

@eddiemuc 
Can you cross-check if I haven't broken anything in `SimpleDialog.input`? Thanks.

eddiemuc commented:
Code is fine. Wrt label (ft in your example) I would propose to use longer unit names now that it is used as a label (e.g. "feet" "kilometer" instead of "ft" / "km").

There are, however, two new lint issues.

eddiemuc reviewed (approved):
Code is fine. Wrt label (ft in your example) I would propose to use longer unit names now that it is used as a label (e.g. "feet" "kilometer" instead of "ft" / "km").

There are, however, two new lint issues.

eddiemuc commented on main/res/layout/dialog_edittext.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
One more thing: I don't think you have to wrap the `TextInputLayout` in another layout here, you might have less problems with programmatic view creation if you just put the `TextInputLayout` here as top-level view (like e.g. done with the `MaterialButton` in `button_view.xml` or `button_icon_view.xml`). Also, such a view snippet is more flexible usabe by other parts of the code later.

moving-bits replied:
Ah, good idea, thanks! I'm so used to having either LinearLayout, FrameLayout or something similar as enclosing frame that I did not try TextInputLayout yet. Will do so, would make things easier.

fm-sys commented:
>> ![image](https://user-images.githubusercontent.com/3754370/122285753-62801980-ceef-11eb-964d-61908663691e.png)




> Wrt label (ft in your example) I would propose to use longer unit names now that it is used as a label (e.g. "feet" "kilometer" instead of "ft" / "km").
> 


Actually, showing the suffix as hint text is kind of ugly in total. There's an official way to set a text prefix and suffix... See:
https://github.com/material-components/material-components-android/issues/1459#issuecomment-813377142
https://github.com/material-components/material-components-android/issues/1427#issuecomment-647103856

moving-bits commented:
> Code is fine. Wrt label (ft in your example) I would propose to use longer unit names now that it is used as a label (e.g. "feet" "kilometer" instead of "ft" / "km").
> 
> There are, however, two new lint issues.

Thanks for taking the time to review. I'll look into the lint issues.

moving-bits commented:
> Actually, showing the suffix as hint text is kind of ugly in total. There's an official way to set a text prefix and suffix... See:
> [material-components/material-components-android#1459 (comment)](https://github.com/material-components/material-components-android/issues/1459#issuecomment-813377142)
> [material-components/material-components-android#1427 (comment)](https://github.com/material-components/material-components-android/issues/1427#issuecomment-647103856)

Ah, that's good to know, thanks for the links. I had searched for something like this, but couldn't find it on the material.io page for text fields. I'll look into the links provided and adapt the input method accordingly.

moving-bits commented:
Updated the PR to use `suffixText` for the units, removed the surrounding `FrameLayout`, and added some color definitions (which helps in legacy settings mode).

Works fine generally, except that if you use a suffix, the textbox changes its height?! A bit strange, but happens in legacy settings context only, does work in regular program context (although not yet used there).

![image](https://user-images.githubusercontent.com/3754370/122436834-6d8f8400-cf99-11eb-9ed4-86d8faca65ea.png).![image](https://user-images.githubusercontent.com/3754370/122436856-741dfb80-cf99-11eb-86e3-96f19d972a9b.png)

moving-bits commented:
failing test is unrelated, so let's merge
